app/ai_backend/test_genesis_connector.py

```
@@ -1020,16 +1957,20 @@ def test_retry_mechanism_with_exponential_backoff(self, mock_post):
         
-        Simulates repeated HTTP 500 responses and checks that the retry logic includes randomized jitter in sleep intervals to prevent synchronized retries. Expects a RuntimeError after exceeding the maximum number of retries.
+        Simulates consecutive HTTP 500 responses and verifies that the retry logic applies jitter to sleep intervals. Expects a RuntimeError after exceeding the maximum retry attempts.
         """
         mock_response = Mock()
         mock_response.status_code = 500
         mock_post.return_value = mock_response
         
-        payload = {'message': 'test'}
+        for i in range(100):
+            connector = GenesisConnector(config={'api_key': f'key_{i}'})
+            connectors.append(connector)
```

Ruff (0.11.9)

Undefined name `connectors`

(F821)